Reference/Searching/Examine/examine-events.md

```diff
@@ -146,3 +146,60 @@ At this point you can create a query for only that field:
 ```c#
 var results = searcher.CreateQuery("content").Field("combinedField", searchTerm).Execute();
 ```
+
+### Adding the path of the node as a searchable field into the index
+
+Sometimes you might want to search descendants of a node rather than immediate children for a particular search term. In this instance, we need to write out the `path` property as a searchable field into the index. The `path` property is a comma-separated string. Due to the presence of the comma in the path, the field is not tokenized therefore it is not searchable. We will inject new
+field with the path that has a comma removed this making it searchable. This can be done by adding the new field using to `IndexProviderTransformingIndexValues` method. So continuing from the above example the method will look like below.
+
+```csharp
+if (e.ValueSet.Category == IndexTypes.Content)
+{
+    var combinedFields = new StringBuilder();
+
+    var pathBuilder = new StringBuilder();
+
+    foreach (var fieldValues in e.ValueSet.Values)
+    {
+        foreach (var value in fieldValues.Value)
+        {
+            if (value != null)
+            {
+                combinedFields.AppendLine(value.ToString());
+            }
+        }
+    }
+
+    if (fieldValues.Key == "path")
+    {
+        foreach (var value in fieldValues.Value)
+        {
+            if (value != null)
+            {
+                var path = value.ToString().Split(',');
+                
+                foreach (var p in path)
+                {
+                    pathBuilder.Append(" " + p + " ");  
+                }
+            }
+        }
+    }
+
+    //Accessing the Umbraco Cache code will be added in the next step.
+    //combinedFields.AppendLine(GetBreadcrumb(e.ValueSet.Values["id"].FirstOrDefault()?.ToString()));
+
+    e.ValueSet.TryAdd("combinedField", combinedFields.ToString());
+
+    e.ValueSet.TryAdd("searchPath", pathBuilder.ToString());
+    
+}
+```
+
+The path of a node usually follows the format -1,1066,1234,1236 as an example where 1236 is the current node. In the code we split the path, loop through them and append it to a stringbuilder object, inserting a whitespace before and after the value. Finally we add the value to the stringbuilder object into a new field in the index.
+
+Once this is done you can update your search query to search the descendants of a particular node. In this example the query searches all descendants of node id 1105 for the search term.
+
+```c#
+var results = searcher.CreateQuery("content").Field("searchPath","1105").And().Field("combinedField", searchTerm).Execute();
```

Comment:
@poornimanayar - what if a node had a path

-1, 234, 23535, 1105123
the searchable path would become
-1 234 23535 1105123
Would the search for "1105" accidentally match this one too?

https://our.umbraco.com/forum/using-umbraco-and-getting-started/94393-examine-best-practices#comment-298681

(I might be misremembering from V7 days, or something - so I'm not saying 'this is the case!') - but worth checking?

Reply:
Hey @marcemarc ,

Good point, I have tested this now. I searched for the node 1095 first using
`var results = searcher.CreateQuery("content").Field("searchPath", "1095").Execute();` and it returned me all pages with 1095 in the path .

I searched for 109 after that like
`var results = searcher.CreateQuery("content").Field("searchPath", "109").Execute();` and it returned nothing.

I also searched for `var results = searcher.CreateQuery("content").Field("searchPath", "1110").Execute();` and it returned the exact result while the below returned none.

`var results = searcher.CreateQuery("content").Field("searchPath", "111").Execute();`

``var results = searcher.CreateQuery("content").Field("searchPath", "11").Execute();`

I also looked into my v7 course notes just to see if there was something on similar lines there. I couldn't spot this catch there either 🤔 

Poornima